### PR TITLE
Fix: Don't open subfolders when using selection rectangle in column view

### LINKED ISSUE #13418: Bug: Issue with selection rectangle in the column layout
### Description

I'm experiencing a problem when using the column view layout. When I try to select multiple folders, the first one unexpectedly opens. This behavior is not desired because when I'm in a subfolder of the folder I want to select, it closes that subfolder and opens the first selected folder. As a result, when I attempt to select multiple folders, the first one opens and then closes when I select an additional folder.

### Steps To Reproduce

1. Switch to the column layout
2. Select multiple folders using the selection rectangle
3. Notice that the first folder that's selected is opened for moment, but then closes after additional folders/items are selected

### Requirements

Delay opening the first folder if selected by the selection rectangle (in the columns layout)

### Files Version

5.5.29.0

### Windows Version

10.0.22621.2283

### Log File

...

### PR DESCRIPTION
**Pull request by heftymouse**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #13418

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   1. Open app ...
   2. Click settings button ...

**Screenshots (optional)**
https://github.com/files-community/Files/assets/59918974/6d086769-1fe8-4e15-a6d4-b3fb88821369

### CODE REVIEW

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 1]
```diff
@@ -36,11 +36,14 @@ public sealed partial class ColumnViewBase : StandardViewBase
...
 		private ListViewItem? openedFolderPresenter;
 
+		private bool isDragging = false;
+
 		public ColumnViewBase() : base()
 		{
 			InitializeComponent();
 			var selectionRectangle = RectangleSelection.Create(FileList, SelectionRectangle, FileList_SelectionChanged);
-			selectionRectangle.SelectionEnded += SelectionRectangle_SelectionEnded;
+			selectionRectangle.SelectionEnded += SelectionRectangle_SelectionEnded1;
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:45]: I think you can use `Override`  instead of manually subscribing.

**Comment by yaira2** [comment]
I still see the sub folder open for a moment before it closes again.

**Comment by Lukiluc29** [comment]
When I select multiple folders, the first folder always opens.

**Comment by yaira2** [comment]
One change I would suggest is that if the user finishes the selection and only one folder is selected, we should open the sub folder for that.

**Comment by Lukiluc29** [comment]
Seems much better to me!

**Comment by Lukiluc29** [comment]
This even addresses this issue: #13600

**Comment by Lukiluc29** [comment]
Why open the latest folder?

**Comment by heftymouse** [comment]
> Why open the latest folder?

You're gonna have to take that up with @yaira2 lol

**Comment by Lukiluc29** [comment]
This patch looks very good to me and effectively fixes the issue.

**Comment by yaira2** [comment]
The behavior looks good. Can you add some code comments so we don't break this in the future?

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 2]
```diff
@@ -482,6 +500,22 @@ protected override void BaseFolderSettings_LayoutModeChangeRequested(object? sen
 			}
 		}
 
+		protected override void SelectionRectangle_SelectionEnded(object? sender, EventArgs e)
+		{
+			isDragging = false;
+			// Open selected folder only if drag resulted in a single folder being selected
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:506]:
```suggestion
			// Open selected folder (as long as a single folder is selected) after the user finishes dragging the selection rectangle
```

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 3]
```diff
@@ -482,6 +500,22 @@ protected override void BaseFolderSettings_LayoutModeChangeRequested(object? sen
...
+			if (SelectedItems?.Count is 1
+				&& SelectedItem is not null
+				&& SelectedItem.PrimaryItemAttribute is StorageItemTypes.Folder)
+				ItemInvoked?.Invoke(new ColumnParam { Source = this, NavPathParam = (SelectedItem is ShortcutItem sht ? sht.TargetPath : SelectedItem.ItemPath), ListView = FileList }, EventArgs.Empty);
+			base.SelectionRectangle_SelectionEnded(sender, e);
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:511]:
```suggestion

			base.SelectionRectangle_SelectionEnded(sender, e);
```

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 4]
```diff
@@ -36,10 +36,16 @@ public sealed partial class ColumnViewBase : StandardViewBase
 
 		private ListViewItem? openedFolderPresenter;
 
+		// Used to disable opening the first folder when selected by the selection rectangle (#13418)
+		// When an item is selected via selection rectangle, the open subfolder is closed, and selected folders are not opened
+		// If only a single folder is selected by the end, it is opened
+		private bool isDragging = false;
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:42]:
```suggestion
		private bool isDraggingSelectionRectangle = false;
```

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 5]
```diff
@@ -211,17 +217,29 @@ protected override void FileList_SelectionChanged(object sender, SelectionChange
 				presenter!.Background = this.Resources["ListViewItemBackgroundSelected"] as SolidColorBrush;
 			}
 
-			if (SelectedItems?.Count == 1 && SelectedItem?.PrimaryItemAttribute is StorageItemTypes.Folder && openedFolderPresenter != FileList.ContainerFromItem(SelectedItem))
+			if (SelectedItems?.Count == 1 && SelectedItem?.PrimaryItemAttribute is StorageItemTypes.Folder)
 			{
-				if (UserSettingsService.FoldersSettingsService.ColumnLayoutOpenFoldersWithOneClick)
+				// Close any open folder
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:222]:
```suggestion
				// Prevents selected folders from opening if the user is currently dragging the selection rectangle (#13418)
```

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 6]
```diff
@@ -211,17 +217,29 @@ protected override void FileList_SelectionChanged(object sender, SelectionChange
...
+				if (isDragging)
+				{
+					CloseFolder();
+					return;
+				}
+
+				if (openedFolderPresenter == FileList.ContainerFromItem(SelectedItem))
+					return;
+
+				// Only open folder if selected through tap
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:232]:
```suggestion
				// Open the selected folder if selected through tap
```

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 7]
```diff
@@ -36,10 +36,16 @@ public sealed partial class ColumnViewBase : StandardViewBase
 
 		private ListViewItem? openedFolderPresenter;
 
+		// Used to disable opening the first folder when selected by the selection rectangle (#13418)
+		// When an item is selected via selection rectangle, the open subfolder is closed, and selected folders are not opened
+		// If only a single folder is selected by the end, it is opened
```
Comment (yaira2) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:41]:
```suggestion
		// Indicates if the selection rectangle is currently being dragged
```

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs (hunk 5 above)
Comment (heftymouse) [reply src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:220]: Can I clarify that it only affects the first selected item?

Comment (yaira2) [reply src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:220]: Either way is fine with me

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by Lukiluc29** [comment]
Thank you very much for this correction.

**Pull request merged**